### ISSUE TITLE
Priority substrings

### ISSUE THREAD

**Issue by asayers** [nixprime/cpsm#7]
`cpsm` does a good job of guessing the right file based on very little information, but the user probably has some additional information which could help it along; for instance:

- I'm more likely to want files with a ".c" extension than ".pcap"
- I'm more likely to want files in "src/" than in "test-data/"
- etc.

I'm not exactly sure what the best way of exploiting this information is. One way would be to read a variable containing a list of exact substrings which, if found, would increase the score of a match. Thoughts?

**Comment by nixprime** [comment]
UI is the trickiest part of a feature like this. The problem is basically that it's not obvious how strongly a priority substring match should be weighted; in fact, I suspect the answer depends on the user and substring. Nor do I want to expose details about cpsm's scoring algorithm in the UI, since they're subject to change.

**Comment by asayers** [comment]
You're right - it's tricky. One simple approach would be to create a partition: all prioritised entries score higher than any non-prioritised entry. My problem, as the examples above may have suggested, is that I sometimes blindly hit enter and get a screen full of binary test data. In my case it would certainly make sense to return non-source files only when all source files have been ruled out.

(I realise I could simply exclude those files from the file list - in my case it would be reasonable - but in general it seems like a fairly drastic option.)

**Comment by LemonBoy** [comment]
It probably makes sense to take into account during the scoring process the `suffixes` option.
